components/prism-pascaligo.js

```diff
@@ -0,0 +1,58 @@
+// Pascaligo is a layer 2 smart contract language for the tezos blockchain
+
+Prism.languages.pascaligo = {
+	'comment': [
+		/\(\*[\s\S]+?\*\)/,
+		/\/\/.*/
+	],
+	'string': [
+		{
+			pattern: /(?:'(?:''|[^'\r\n])*'|#[&$%]?[a-f\d]+)+|\^[a-z]/i,
+			greedy: true
+		},
+		{
+			pattern: /(?:"(?:""|[^"\r\n])*"|#[&$%]?[a-f\d]+)+|\^[a-z]/i,
+			greedy: true
+		},
+	],
+	'keyword': [
+		{
+			pattern: /(^|[^&])\b(?:if|while|for|return|nil|remove|from|else|then|skip|with|is|function|var|const|type|end|begin|block)\b/i,
+			lookbehind: true
+		},
+		{
+			pattern: /(^|[^&])\b(?:True|False)\b/i,
+			lookbehind: true
+		},
+    ],
+    'builtin': {
+        pattern: /(^|[^&])\b(?:int|unit|string|nat|map)\b/i,
+		lookbehind: true
+    },
+    'function': {
+        pattern: /(?!with\b)\b\w+ ?(?=\()/i,
+    },
+    'variable': [
+        { 
+            pattern: /\w+ ?(?=\:\=)/i,
+        },
+        { 
+            pattern: /\w+ ?(?=\:)/i,
+        },
+    ],
+    'class-name': {
+        pattern: /\w+ ?(?= is ?)/i,
```

Comment:
A word about spaces: We generally opt to not included leading or trailing spaces inside tokens. Also, just matching normal spaces is usually not enough which is why we usually use `\s` (which also includes tabs and newlines) for spaces in C style languages.
This also applies to the `function` and `variable` patterns.

```suggestion
        pattern: /\w+(?=\s+is\b)/i,
```